Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Controllers/ProductController.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Know_Your_Nation_Speedy.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace Know_Your_Nation_Speedy.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProductController : ControllerBase
+    {
+        private readonly MyDbContext _db;
+        public ProductController(MyDbContext context)
+        {
+            _db = context;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Product>>> Get()
+        {
+            return await _db.ProductEntries.ToListAsync();
+        }
+ 
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetEntry([FromRoute] int id)
+        {
+            var entry = await _db.ProductEntries.SingleOrDefaultAsync(o => o.Id == id);
+            if (entry == null)
+            {
+                return NotFound();
+            }
+            await _db.SaveChangesAsync();
+            return Ok(entry);
+        }
+
+
+        [HttpPost]
+        public async Task Post([FromBody] Product product)
+        {
```

Comment:
You need to do a modelstate check here and return a `BadRequest();` if its invalid